sycl/include/sycl/ext/oneapi/device_global/device_global.hpp

```diff
@@ -42,9 +42,28 @@ struct HasArrowOperator<
 template <typename T, typename PropertyListT, typename = void>
 class device_global_base {
 protected:
-  T *usmptr;
-  T *get_ptr() noexcept { return usmptr; }
-  const T *get_ptr() const noexcept { return usmptr; }
+  using pointer_t = typename multi_ptr<T, access::address_space::global_space,
```

Comment:
Nit: there is `sycl::decorated_global_ptr` alias, I suppose it is shorter to use